test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +214,45 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 void is_about(float x, float y)
 {
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
-
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
-
-CLANG_DIAGNOSTIC_POP
+    if (x != y)
+        assert(std::abs((x-y)/(x+y)) < 1.e-6);
 }
 
 void is_about(double x, double y)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    if (x != y)
+        assert(std::abs((x-y)/(x+y)) < 1.e-14);
 }
 
 void is_about(long double x, long double y)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    if (x != y)
+        assert(std::abs((x-y)/(x+y)) < 1.e-14);
+}
+
+template <typename T1, typename T2>
+void
+is_about(const dpl::complex<T1>& x, const dpl::complex<T2>& y)
```

Comment:
I think such comparison is not correct for complex number...
For more details see https://stackoverflow.com/questions/34231008/how-to-compare-two-complex-numbers
or https://math.stackexchange.com/questions/310931/comparing-complex-numbers

Reply:
fixed